src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Marshaling/JSMarshalerArgument.Task.cs

```diff
@@ -140,13 +140,21 @@ internal void ToJSDynamic(Task? value)
         {
             Task? task = value;
 
+            var ctx = ToJSContext;
+
+            var isCurrentThreadOrParameter = ctx.IsCurrentThread() || slot.Type != MarshalerType.TaskPreCreated;
```

Comment:
Can we have a comment describing the parameter, like in the PR description? For me "non-return, JSImport" part is not clear without it.
```suggestion
            // current thread or non-return parameter of JSImport
            var isCurrentThreadOrParameter = ctx.IsCurrentThread() || slot.Type != MarshalerType.TaskPreCreated;
```

Reply:
I refactored it into `CanMarshalTaskResultOnSameCall` with detailed explanation